Lib/socketserver.py

```diff
@@ -474,6 +477,14 @@ def server_activate(self):
         """
         self.socket.listen(self.request_queue_size)
 
+    def server_shutdown(self):
+        """Called to shutdown and close the server."""
+        try:
+            self.socket.shutdown(socket.SHUT_RDWR)
+        except OSError:
+            pass  # Don't call shutdown on a bad file descriptor
+        self.server_close()
```

Comment:
This method mark as overridden but if we override it we need to know that need to call `self.server_close()`. This is not obvious in my opinion.

Reply:
Fixed, from now server_shutdown will only shutdown the socket without closing it.